webview/WebView2_spec.md

```diff
@@ -0,0 +1,247 @@
+await myWebView2.ExecuteScriptAsync(new string[] { functionString });
+}
+```
+
+<br>
+
+**WebMessageReceived:**
+Scripts in the WebView2 content can use window.chrome.webview.postMessage.postMessage(“Message”) with JSON parameters to send information back to your app. To receive these messages, handle the WebMessageReceived event.
+
+
+```
+// Event allows for notification when a webMessage event is fired from the JavaScript on the html side of the webview.
+
+myWebView2.WebMessageReceived += (WebView2 sender, WebView2WebMessageReceivedEventArgs args) =>
+{
+    // Important to validate that the Uri is what we expect from that webview.
+    string uriAsString = sender.UriSource.ToString();
+
+    if (args.Source == uriAsString)
+    {
+        HandleWebMessageAsString(args.WebMessageAsString);
+        HandleWebMessageAsJson(args.WebMessageAsJson);
+    }
+
+    else
+    {
+        // If the source is not validated, don't process the message.
+        return;
+    }
+};
+```
+<br>
+
+**NavigationStarting:**
+Occurs before the WebView2 navigates to new content. You can cancel navigation in a handler for this event by setting the WebView2NavigationStartingEventArgs.Cancel property to true.
+
+
+```
+myWebView.NavigationStarting += myWebView_NavigationStarting;
+
+private void myWebView_NavigationStarting(WebView2 sender, WebView2NavigationStartingEventArgs args)
+{
+    if (!IsAllowedUri(args.Uri))
+    {
+        args.Cancel = true;
+    }
+}
+```
+<br>
+
+**ContentLoading:**
+ContentLoading fires when new content has started loading on the webview's main frame or if a same page navigation occurs (such as through fragment navigations or history.pushState navigations). This follows the NavigationStarting event and precedes the NavigationCompleted event.
+
+<br>
+
+**SourceChanged:**
+Occurs when the 'Source' property is changed
+
+<br>
+
+**NavigationCompleted**:
+Occurs when the WebView2 has finished loading the current content or if the navigation has failed. To determine whether the navigation has failed, check the IsSuccess and WebErrorStatus properties of the WebViewNavigationCompletedEventArgs class.
+
+
+```
+myWebView2.NavigationCompleted += myWebView2_NavigationCompleted;
+
+private void myWebView2_NavigationCompleted(WebView2 sender, WebView2NavigationCompletedEventArgs args)
+{
+    if (args.IsSuccess == true)
+    {
+        statusTextBlock.Text = "Navigation to “ +
+        args.Uri.ToString() + " completed successfully.";
+    }
+
+    else
+    {
+    statusTextBlock.Text = "Navigation to: " + args.Uri.ToString() + " failed with error ” + args.WebErrorStatus,ToString();
+    }
+}
+```
+
+<br>
+
+**Reload:**
+Fires navigation event without altering the navigation stack.
+
+<br>
+
+# Remarks
+
+// This API may ship before the WinRT CoreWebView object is included, meaning functionality of the Webview2 will be limited to the APIs included on this object.
```

Comment:
Please put this in the Background section; Remarks is intended to go into the Remarks section of the docs.microsoft.com page.